cmd/kubeadm/app/util/logfile.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"io/ioutil"
+	"os"
+)
+
+// kubeadm record file path and name define
+const (
+	KubeadmLogFolder = "/var/lib/kubeadm"
+	KubeadmLogFile   = KubeadmLogFolder + "/" + "kubeadm.log"
+)
+
+// WriteKubeadmLogFile write record to kubeadm log file when kubeadm init or kubeadm join
+func WriteKubeadmLogFile(content string) error {
+	const kubeadmLogPermissions = 0666
+	s := []byte(content)
+	if _, err := os.Stat(KubeadmLogFolder); err != nil {
+		err := os.MkdirAll(KubeadmLogFolder, kubeadmLogPermissions)
+		if err != nil {
+			return err
+		}
+	}
+	if _, err := os.Stat(KubeadmLogFile); err == nil {
+		return nil
+	}
+	if err := ioutil.WriteFile(KubeadmLogFile, s, kubeadmLogPermissions); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ReadKubeadmLogFile reads the log file from disk and determines if `init` or `join` ran on this machine.
+func ReadKubeadmLogFile() (string, error) {
```

Comment:
same for reading:
```
func ReadKubeadmLogFile() error {
	return readKubeadmLogFile(path.Join(kubeadmLogFolder, kubeadmLogFile))
}

func readKubeadmLogFile(filePath string) error {
	data, err := ioutil.ReadFile(filePath)
	....
}
```

Reply:
need return (string error)
func ReadKubeadmLogFile()(string error) {

Reply:
that's true. missed it.